### PR TITLE
initial request implementation

### PR DESCRIPTION
**Pull request by howardchung**
Requests page
Distributions page

### CODE REVIEW

**Comment by howardchung** [comment]
any comments @mike-robertson ?

File: components/Distributions/Distributions.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+              }
+              return value;
+            },
+          },
+        },
+      };
+      c3.generate(options);
+    }
+  }
+  render() {
```
Comment (mike-robertson) [inline components/Distributions/Distributions.jsx]: You're not using redux?

**Comment by mike-robertson** [comment]
Only comment is you're no longer using redux. I'd prefer we stick to that for state management so we can have a Flux pattern and be consistent.

File: components/Distributions/Distributions.jsx (hunk 1 above)
Comment (howardchung) [inline components/Distributions/Distributions.jsx]: We could move it later.  For now, though, this doesn't rely on any global state, and it's less code, so quicker to prototype.

Comment (mike-robertson) [inline components/Distributions/Distributions.jsx]: The point of redux isn't to use it when it relies on global state. It's to force a single source of truth for all state which is immutable from the components' perspective. I'm against merging this for speed reasons, it's not like these are hard use cases for redux, all our stuff has been this exact scenario, so you should be able to base it off existing patterns in the code.

Comment (howardchung) [inline components/Distributions/Distributions.jsx]: OK I'll make actions and reducers for these

**Comment by howardchung** [comment]
@mike-robertson can you re-review?

File: yasp.config.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
-  API_HOST: 'https://yasp.co',
-  // API_HOST: 'http://hchung.westus.cloudapp.azure.com:5000',
+  // API_HOST: 'https://yasp.co',
```
Comment (howardchung) [inline yasp.config.js]: undo before checkin

File: actions/distributionsActions.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import fetch from 'isomorphic-fetch';
+import { API_HOST } from '../actions';
+
+const url = '/api/distributions';
+
+const START = 'yasp/distributions/START';
+const DONE = 'yasp/distributions/DONE';
+const ERROR = 'yasp/distributions/ERROR';
+
```
Comment (mike-robertson) [inline actions/distributionsActions.js]: we should be consistent with these action names. Can you name them similar to other async requests?

File: actions/requestActions.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  const formData = new FormData();
+  formData.append('match_id', matchId);
+  dispatch(requestStart());
+  return fetch(`${API_HOST}${url}`, {
+    method: 'post',
+    body: formData,
+  })
+  .then(res => res.json())
+  .then((json) => {
+    function poll() {
```
Comment (mike-robertson) [inline actions/requestActions.js]: don't define this function inside of your `then()`. Just declare it above the method and then call it with dispatch as an argument. This way we aren't defining this function every time you dispatch this action, and it's also easier to read.

**Comment by mike-robertson** [comment]
Only comments are one where to define that polling function and naming consistency with actions for requests. Other than that I think it's good to go.

File: actions/distributionsActions.js (hunk 3 above)
Comment (howardchung) [inline actions/distributionsActions.js]: ok, also renamed the search ones.  requestRequest is a bit weird, but i guess consistency is more important.

**Pull request merged**